contracts/BaseWrapper.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {RegistryAPI, VaultAPI} from "./BaseStrategy.sol";
+
+abstract contract BaseWrapper {
+    using SafeMath for uint256;
+
+    ERC20 public token;
+
+    // Reduce number of external calls (SLOADs stay the same)
+    VaultAPI[] private _cachedVaults;
+
+    RegistryAPI public registry;
+
+    constructor(address _token) public {
+        token = ERC20(_token);
+        // v2.registry.ychad.eth
+        registry = RegistryAPI(0xE15461B18EE31b7379019Dc523231C57d1Cbc18c);
+    }
+
+    function _setRegistry(address _registry) internal {
+        // In case you want to override the registry instead of re-deploying
+        registry = RegistryAPI(_registry);
```

Comment:
is my understanding correct that the affiliate is the full admin party on this token (not yearn) since they have full access to swap registry potentially sending deposits to a vault not controlled by yearn?

Or should yearn have some role within the access for these affiliateToken?

Reply:
My position was that the affiliate has full control to do whatever they want, using data (managed by us) from our Registry (e.g. deployment history) and Vaults (depositor controls). I figured if they wanted to delegate additional authority to us, that's their prerogative, not ours to enforce.